dev/mp-dev.sh

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env bash
+set -euo pipefail
+# Uncomment for debugging
+# set -x
+
+# Default environment variables
+: "${MOUNTPOINT_CSI_DEV_REGION:=eu-north-1}"
+: "${MOUNTPOINT_CSI_DEV_ECR_REPO_NAME:=mp-dev}"
+: "${MOUNTPOINT_CSI_DEV_CLUSTER_NAME:=mp-dev-cluster}"
+
+usage() {
+    echo "usage: $0 <command>"
+    echo ""
+    echo "available commands:"
+    echo "  setup                    create ECR repository, EKS cluster, and deploy everything"
+    echo "  teardown                 delete ECR repository and EKS cluster"
+    echo "  deploy-helm-chart        deploy the Helm chart from source"
+    echo "  deploy-containers        build the containers from source, push to ECR, and restart the Csi Driver pods"
+    echo "                              --build-mountpoint: use Dockerfile.local instead of Dockerfile to build Mountpoint from source"
+    echo "  deploy                   deploy both Helm chart and containers from source"
+    echo "  help                     show this help message"
+    echo ""
+    echo "optional environment variables (with defaults):"
+    echo "  MOUNTPOINT_CSI_DEV_REGION          AWS region for the dev stack (default: eu-north-1)"
+    echo "  MOUNTPOINT_CSI_DEV_ECR_REPO_NAME   ECR repository name (default: mp-dev)"
+    echo "  MOUNTPOINT_CSI_DEV_CLUSTER_NAME    EKS cluster name (default: mp-dev-cluster)"
+    exit 1
+}
+
+validate_env_vars() {
+    local missing_vars=()
+    local required_vars=("$@")
+
+    for var in "${required_vars[@]}"; do
+        if [[ -z "${!var:-}" ]]; then
+            missing_vars+=("$var")
+        fi
+    done
+
+    if [[ ${#missing_vars[@]} -gt 0 ]]; then
+        echo "error: missing required environment variables:"
+        for var in "${missing_vars[@]}"; do
+            echo "  $var"
+        done
+        echo ""
+        exit 1
+    fi
+}
+
+get_aws_account_id() {
+    aws sts get-caller-identity --query Account --output text
+}
+
+get_ecr_repository_url() {
+    local repo_name="${MOUNTPOINT_CSI_DEV_ECR_REPO_NAME}"
+    local region="${MOUNTPOINT_CSI_DEV_REGION}"
```

Comment:
nit: can we avoid the redeclaration of these two in all ecr repository related functions?